lib/screens/landing/login_screen.dart

```diff
@@ -24,32 +39,107 @@ class _LoginScreenState extends State<LoginScreen> {
   }
 
   void _userLogin() async {
-    const String apiUrl = 'http://localhost:8080/users/login';
+    if (emailController.text.isEmpty) {
+      Fluttertoast.showToast(
+        msg: "Please enter your email or username",
+        toastLength: Toast.LENGTH_SHORT,
+        gravity: ToastGravity.CENTER,
+        timeInSecForIosWeb: 1,
+        backgroundColor: Colors.grey,
+        textColor: Colors.white,
+      );
+      return;
+    }
+
+    if (passwordController.text.isEmpty) {
+      Fluttertoast.showToast(
+        msg: "Please enter your password",
+        toastLength: Toast.LENGTH_SHORT,
+        gravity: ToastGravity.CENTER,
+        timeInSecForIosWeb: 1,
+        backgroundColor: Colors.grey,
+        textColor: Colors.white,
+      );
+      return;
+    }
+
+    setState(() {
+      isLoading = true;
+      searchResult = {};
+    });
+
+    const String apiUrl_email = 'http://localhost:8080/users/login';
+    const String apiUrl_username = 'http://localhost:8080/users/username/login';
 
```

Comment:
why is there two calls to our backend just to login? should try to combine the logic into just one call for a user to login. probs need to discuss with @Dongchunhua1225 on this

Reply:
one is for username and the other one is for email, the user can log in if one of them works

Reply:
ohh got it, so it will just be one api call anyways? since user can input either just username or email